From a2cc5f66ed4c5d569b5cf6a2f3f3eed9a540e695 Mon Sep 17 00:00:00 2001
From: Vu Dang <vu.dang.te@renesas.com>
Date: Mon, 2 Oct 2023 09:30:29 +0700
Subject: [PATCH 5/7] kick cm33 before loading kernel

There is a change in loading images sequence.
Must kick CM33 before loading Kernel image to save booting time

Signed-off-by: Vu Dang <vu.dang.te@renesas.com>
---
 bl2/bl2_image_load_v2.c | 15 ++++++++++++++-
 bl2/bl2_main.c          |  6 +++---
 2 files changed, 17 insertions(+), 4 deletions(-)

diff --git a/bl2/bl2_image_load_v2.c b/bl2/bl2_image_load_v2.c
index dee3fc2d3..d3e274570 100644
--- a/bl2/bl2_image_load_v2.c
+++ b/bl2/bl2_image_load_v2.c
@@ -18,6 +18,9 @@
 
 #include <platform_def.h>
 
+extern void init_I2C_clk(void);
+extern void kick_cm33();
+
 /*******************************************************************************
  * This function loads SCP_BL2/BL3x images and returns the ep_info for
  * the next executable image.
@@ -65,7 +68,7 @@ struct entry_point_info *bl2_load_images(void)
 
 		if ((bl2_node_info->image_info->h.attr &
 		    IMAGE_ATTRIB_SKIP_LOADING) == 0U) {
-			INFO("BL2: Loading image id %u\n", bl2_node_info->image_id);
+			NOTICE("BL2: Loading image id %d\n", bl2_node_info->image_id);
 			err = load_auth_image(bl2_node_info->image_id,
 				bl2_node_info->image_info);
 			if (err != 0) {
@@ -84,6 +87,16 @@ struct entry_point_info *bl2_load_images(void)
 			plat_error_handler(err);
 		}
 
+		if(bl2_node_info->image_id == BL331_IMAGE_ID)
+		{
+			NOTICE("BL2: Kich CM33 image id = %d\n", bl2_node_info->image_id);
+			/* init_I2C_clk */
+			init_I2C_clk();
+
+			/* Kick CM33 */
+			kick_cm33();
+		}
+
 		/* Go to next image */
 		bl2_node_info = bl2_node_info->next_load_info;
 	}
diff --git a/bl2/bl2_main.c b/bl2/bl2_main.c
index fe5c6e836..894b60760 100644
--- a/bl2/bl2_main.c
+++ b/bl2/bl2_main.c
@@ -222,11 +222,11 @@ void bl2_main(void)
 	/* Load the subsequent bootloader images. */
 	next_bl_ep_info = bl2_load_images();
 
-		/* init_I2C_clk */
-		init_I2C_clk();
+	/* init_I2C_clk */
+	//init_I2C_clk();
 
         /* Kick CM33 */
-        kick_cm33();
+        //kick_cm33();
 
         /* Teardown the Measured Boot backend */
 	bl2_plat_mboot_finish();
-- 
2.34.1

